napari/layers/image/image.py

```diff
@@ -689,3 +689,41 @@ def contrast_limits(self, contrast_limits):
                 self.refresh()
             finally:
                 self._keep_auto_contrast = prev
+
+    def _calculate_value_from_ray(self, values):
+        values_within_clim = (values >= self.contrast_limits[0]) & (
+            values <= self.contrast_limits[1]
+        )
+        values_masked = np.where(values_within_clim, values, np.nan)
```

Comment:
I don't think this is how this should work at all??? The values should be *clipped* to the contrast limit, not masked! And it's kind of a toss up for me whether the clipping should happen *before* walking the ray or after.

For example, if I'm doing a maximum intensity projection with contrast limits [0, 5] and the ray is:

[0, 3, 4, 5, 7, 9, 4, 0]

I want either the (position of the) 5 or the (position of the) 9. (It's a toss up, but for me I'd rather the 9. But it's more accurate to say it's the 5 when it comes to what VisPy does.) But I *certainly* don't want the position of the 4!

If you look at the 2D behaviour, we always return the true value under the cursor, regardless of contrast limits. So I think that we should return 9 in this case.

Reply:
Mhm, good point!

And yeah, this goes back to the discussion of "we should return the real value, not the rendered value".

Reply:
At the same time, it seems reasonable to return nothing if all you see is black... so I guess the first check should stay as is.